File: Python3/프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    D = [(progresses[i]-100)//speeds[i] * (-1) for i in range(len(speeds))]
    i = 0
    j = 1
    while i < len(D):
        while j < len(D) and D[i] >= D[j]:
            j += 1
            if j == len(D):
                break
        answer.append(j-i)
        i = j
        j = i+1
    return answer

File: Java/프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public Deque<Integer> solution(int[] progresses, int[] speeds) {
        Deque<Integer> P = new ArrayDeque<>();
        Deque<Integer> S = new ArrayDeque<>();
        Deque<Integer> A = new ArrayDeque<>();
        
        for (int i = 0; i < progresses.length; i++){
            P.offer(progresses[i]);
            S.offer(speeds[i]);
        }
        
        while(!P.isEmpty()){
            int n = P.size();
            int x = 0;
            for (int i = 0; i < n; i++){
                int p = P.poll();
                int s = S.poll();
                P.offer(p+s);
                S.offer(s);
            }
            for (int i = 0; i < n; i++){
                int p = P.getFirst();
                if (p < 100){
                    break;
                }
                P.poll();
                S.poll();
                x++;
            }
            if (x > 0){
                A.offer(x);
            }
        }

        return A;
    }
}